### PR TITLE
Navigate directly to the backup flow if there is only one not backed up wallet group

### PR DESCRIPTION
**Pull request by vyorkin**

### CODE REVIEW

File: src/ui/shared/requests/useNotBackedUpWalletGroups.ts [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    useErrorBoundary: true,
+    select: options?.select,
+  });
+}
+
+export function useNotBackedUpWalletGroupsCount() {
+  const { data: count } = useNotBackedUpWalletGroups({
+    select: (walletGroups) => walletGroups.length,
+  });
+  return count ?? 0;
```
Comment (zerts) [inline src/ui/shared/requests/useNotBackedUpWalletGroups.ts:22]: Can't we just return data.length?

File: src/ui/shared/channels.mock.ts [hunk 2]
```diff
@@ -111,7 +111,7 @@ class WalletPortMock {
         mockRecord.walletManager.groups
       );
       return result;
-    } else if (method === 'getNoBackupCount') {
+    } else if (method === 'getNotBackedUpWalletGroups') {
```
Comment (zerts) [inline src/ui/shared/channels.mock.ts:114]: The result of this case should not be a number any more

File: src/ui/shared/requests/useNotBackedUpWalletGroups.ts (hunk 1 above)
Comment (zerts) [reply src/ui/shared/requests/useNotBackedUpWalletGroups.ts:22]: Or we can just remove this hook and use the other one in Settings counter. Just add .length in that hook

File: src/ui/pages/Overview/BackupReminder.tsx [hunk 3]
```diff
@@ -74,9 +82,18 @@ function BackupReminderComponent({ onDismiss }: { onDismiss: () => void }) {
...
+            onClick={(event) => {
+              handleHide();
+              if (notBackedUpWalletGroups.length === 1) {
+                openHrefInTabView(event);
+              }
+            }}
             as={UnstyledLink}
-            to="/settings"
+            to={
+              notBackedUpWalletGroups.length > 1
```
Comment (zerts) [inline src/ui/pages/Overview/BackupReminder.tsx:93]: May be it is a bit safer to swap condition parts. So we will not fall into the second line if length === 0

Comment (vyorkin) [reply src/ui/pages/Overview/BackupReminder.tsx:93]: yeah, I think its would be even better to check for `=== 1` ✅

**Review by zerts — APPROVED** [review]

File: src/ui/pages/Overview/BackupReminder.tsx [hunk 4]
```diff
@@ -34,6 +37,13 @@ function BackupReminderComponent({ onDismiss }: { onDismiss: () => void }) {
     dialogRef.current?.close();
   });
 
+  const { data: walletGroups, isLoading } = useWalletGroups();
+  const notBackedUpGroups = walletGroups?.filter((group) => needsBackup(group));
```
Comment (everdimension) [inline src/ui/pages/Overview/BackupReminder.tsx:41]: useMemo

**Review by everdimension — CHANGES_REQUESTED** [review]

File: src/ui/shared/requests/useNotBackedUpWalletGroups.ts (hunk 1 above)
Comment (vyorkin) [reply src/ui/shared/requests/useNotBackedUpWalletGroups.ts]: ✅

File: src/ui/pages/Overview/BackupReminder.tsx (hunk 4 above)
Comment (vyorkin) [reply src/ui/pages/Overview/BackupReminder.tsx]: ✅

**Review by everdimension — APPROVED** [review]

**Pull request merged**